validator/project_validator.go

```diff
@@ -2347,6 +2348,32 @@ func checkTasks(project *model.Project) ValidationErrors {
 	return errs
 }
 
+// checkTaskUsage returns a notice for each task that is defined but unused by any (un-disabled) variant.
+// TODO: upgrade to a warning in DEVPROD-8154
```

Comment:
I thought there were some projects that define tasks but don't use them directly until generate.tasks runs. Is it okay to warn for those projects?